phpunit6 support

omissis opened the pull request:
This PR introduces phpunit6 support and adds PSR-2 compliance.

omissis committed:
port code and tests to phpunit6.

omissis committed:
add psr-2 checks and fixes.

omissis committed:
update readme.

omissis committed:
extend PSR-2 to tests, make phpunit property detection a bit more flexible and up-to-date.

padraic commented on README.md:
```diff
@@ -6,19 +6,13 @@ Inspired by [Kris Wallsmith faster PHPUnit article](http://kriswallsmith.net/pos
 
-    curl -s http://getcomposer.org/installer | php
-    php composer.phar install
+    composer install
```
`composer update`?

omissis replied:
+1

padraic reviewed (approved):
Thanks @omissis, I'll merge and tag as a 2.0 when I can.

omissis commented on composer.json:
```diff
@@ -17,13 +17,19 @@
     },
     "replace": {
-        "mybuilder/phpunit-accelerator": "1.0.x"
+        "mybuilder/phpunit-accelerator": "2.0.x"
```
@padraic I am not sure this is really meaningful, as the original accelerator does not have a 2.0 nor support phpunit6. any recommendation on how to proceed here?

padraic replied:
I'd remove it altogether in that case.

omissis committed:
remove replace section in composer.json as there is no version of the package to replace.

omissis committed:
add type hinting and docblocks.

omissis committed:
replace composer install with composer update in travis config file.

omissis committed:
change dist from ubuntu precise to trusty to keep hhvm compatibility.

omissis committed:
drop support for hhvm, as phpunit does not support it anymore. add allow_failure for php nightly build.

omissis committed:
add fast finish to the travis build matrix

omissis commented:
@padraic ok, I have implemented the two CR comments, then added some type-hinting and docblocks, and reworked a bit the travis config. looks good to me now :)

omissis committed:
fix composer usage and docs.

omissis commented:
I think this has become old enough not to be relevant anymore, closing.
